r2i/onnxrt_openvino/engine.cc

```diff
@@ -0,0 +1,61 @@
+/* Copyright (C) 2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+ */
+
+#include "r2i/onnxrt_openvino/engine.h"
+
+#include <core/providers/openvino/openvino_provider_factory.h>
+
+static const OrtApi *g_ort = OrtGetApiBase()->GetApi(ORT_API_VERSION);
+
+namespace r2i {
+namespace onnxrt_openvino {
+
+Engine::Engine () : onnxrt::Engine() {
+  hardware_option = "CPU_FP32";
+}
+
+void Engine::AppendSessionOptionsExecutionProvider(Ort::SessionOptions
+    &session_options, r2i::RuntimeError &error) {
+  OrtStatus *status = NULL;
+
```

Comment:
clean the error first.

Reply:
why isn't the error returned?

Reply:
I decided to pass r2i::RuntimeError as reference just to be consistent in the code for the base engine implementation. Basically I prefer this:

`  try {`
`    this->CreateEnv();`
`    this->CreateSessionOptions();`
`    this->AppendSessionOptionsExecutionProvider(this->session_options, error);`
`    this->CreateSession((void *) this->model->GetOnnxrtModel().get(),`
`                        this->model->GetOnnxrtModelSize(), error);`
`    this->num_input_nodes = this->GetSessionInputCount(this->session, error);`
`    this->num_output_nodes = this->GetSessionOutputCount(this->session, error);`
`  }`

instead of this:

`  try {`
`    this->CreateEnv();`
`    this->CreateSessionOptions();`
`    error = this->AppendSessionOptionsExecutionProvider(this->session_options);`
`    this->CreateSession((void *) this->model->GetOnnxrtModel().get(),`
`                        this->model->GetOnnxrtModelSize(), error);`
`    this->num_input_nodes = this->GetSessionInputCount(this->session, error);`
`    this->num_output_nodes = this->GetSessionOutputCount(this->session, error);`
`  }`